support AWS CodeArtifact

AWS CodeArtifact  generates the `NPM_REGISTRY` with a trailing `/`:

<img width="801" alt="圖片" src="https://user-images.githubusercontent.com/278432/84459157-54266e80-ac99-11ea-8989-b607d9f0ae0c.png">


And the `jsii-release-npm` will generate `NPM_REGISTRY` with double slash in `.npmrc`

https://github.com/eladb/jsii-release/blob/4d05bf42a90e71dc170bd189bf708234a2cece00/bin/jsii-release-npm#L25

which returns the **Unable to authenticate** error

```bash
$ NPM_REGISTRY='pahud-112233445566.d.codeartifact.ap-northeast-1.amazonaws.com/npm/aws-repo/' NPM_TOKEN=`aws codeartifact get-authorization-token --domain pahud --domain-owner 903779448426 --query authorizationToken --output text` node_modules/jsii-release/bin/jsii-release-npm
```

```
npm notice 
npm notice 📦  cdk-codeartifact-demo@0.0.0
npm notice === Tarball Contents === 
npm notice 19.3kB .jsii                        
npm notice 11.4kB LICENSE                      
npm notice 1.2kB  .projenrc.js                 
npm notice 3.8kB  lib/index.js                 
npm notice 1.7kB  test/index.test.js           
npm notice 1.0kB  test/integ.api.js            
npm notice 1.3kB  .eslintrc.json               
npm notice 2.4kB  package.json                 
npm notice 19B    version.json                 
npm notice 1.6kB  README.md                    
npm notice 346B   lib/index.d.ts               
npm notice 31B    test/index.test.d.ts         
npm notice 11B    test/integ.api.d.ts          
npm notice 397B   .mergify.yml                 
npm notice 339B   .github/workflows/build.yml  
npm notice 1.3kB  .github/workflows/release.yml
npm notice === Tarball Details === 
npm notice name:          cdk-codeartifact-demo                   
npm notice version:       0.0.0                                   
npm notice package size:  12.6 kB                                 
npm notice unpacked size: 46.1 kB                                 
npm notice shasum:        eee847e49c128493f884a17e8bf238d500c653d2
npm notice integrity:     sha512-XH+b7+/PKCedZ[...]wsLe6ajbdlCVw==
npm notice total files:   16                                      
npm notice 
npm ERR! code E401
npm ERR! Unable to authenticate, need: Bearer realm="pahud/aws-repo", Basic realm="pahud/aws-repo"

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/pahud/.npm/_logs/2020-06-12T02_32_35_628Z-debug.log
ERROR
```

A quick fix is to always remove the trailing slash with bash substitution like this

```
echo "//${NPM_REGISTRY%%/}/:_authToken=${NPM_TOKEN}" > ~/.npmrc
```
And it will work with both AWS CodeArtifact as well as npmjs.

--- a/bin/jsii-release-npm
+++ b/bin/jsii-release-npm
@@ -22,7 +22,7 @@ if [ -z "${NPM_TOKEN:-}" ]; then
 fi
 
 NPM_REGISTRY=${NPM_REGISTRY:-"registry.npmjs.org"}
-echo "//${NPM_REGISTRY}/:_authToken=${NPM_TOKEN}" > ~/.npmrc
+echo "//${NPM_REGISTRY%%/}/:_authToken=${NPM_TOKEN}" > ~/.npmrc
 
 # this overrides any registry configuration defined externally. For example, yarn sets the registry to the yarn proxy
 # which requires `yarn login`. but since we are logging in through ~/.npmrc, we must make sure we publish directly to npm.

